### PR TITLE
ensure webdriveragent logs come out line-by-line (fix #6749)

### PR DESCRIPTION
**Pull request by jlipps**
cc @imurchie @yaphatak

### CODE REVIEW

File: lib/webdriveragent.js [hunk 1]
```diff
@@ -182,7 +182,9 @@ class WebDriverAgent {
         return;
       }
       if (isPertinentLogLine(out)) {
-        agentLog.debug(`${prefix}: ${out.trim()}`);
+        for (let line of out.split("\n")) {
+          agentLog.debug(`${prefix}: ${line.trim()}`);
```
Comment (imurchie) [inline lib/webdriveragent.js]: Splitting on `\n` gets rid of the newline, so I don't think we ought to do the `trim()` here. With it everything will be left-justified, and merge together. Without it the indentation that the iOS log produces will be maintained.

```
dbug WebDriverAgent Sim: Aug 18 17:53:59 isaac CoreSimulatorBridge[35498]: Requesting installation of file:///Users/isaac/Library/Developer/Xcode/DerivedData/WebDriverAgent-eoyoecqmiqfeodgstkwbxkfyagll/Build/Products/Debug-iphonesimulator/WebDriverAgentRunner-Runner.app/ with options: {
dbug WebDriverAgent Sim: CFBundleIdentifier = "com.apple.test.WebDriverAgentRunner-Runner";
dbug WebDriverAgent Sim: PackageType = Developer;
dbug WebDriverAgent Sim: SimulatorRootPath = "/Applications/Xcode-7.3.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk";
dbug WebDriverAgent Sim: SimulatorUserPath = "/Users/isaac/Library/Developer/CoreSimulator/Devices/F07AB371-A73B-4C3F-9A85-CD48F9986B61/data";
dbug WebDriverAgent Sim: }
```
vs
```
dbug WebDriverAgent Sim: Aug 18 17:54:57 isaac CoreSimulatorBridge[35798]: Requesting installation of file:///Users/isaac/Library/Developer/Xcode/DerivedData/WebDriverAgent-eoyoecqmiqfeodgstkwbxkfyagll/Build/Products/Debug-iphonesimulator/WebDriverAgentRunner-Runner.app/ with options: {
dbug WebDriverAgent Sim: 	    CFBundleIdentifier = "com.apple.test.WebDriverAgentRunner-Runner";
dbug WebDriverAgent Sim: 	    PackageType = Developer;
dbug WebDriverAgent Sim: 	    SimulatorRootPath = "/Applications/Xcode-7.3.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk";
dbug WebDriverAgent Sim: 	    SimulatorUserPath = "/Users/isaac/Library/Developer/CoreSimulator/Devices/F07AB371-A73B-4C3F-9A85-CD48F9986B61/data";
dbug WebDriverAgent Sim: 	}
```

Comment (imurchie) [inline lib/webdriveragent.js]: I will add that we should probably also check if there is anything in the line. We sometimes get empty lines, which do not need to be printed.

**Comment by imurchie** [comment]
One further point I thought of this morning: we should note if Appium was invoked with `--show-ios-log` (in the driver, if `this.opts.showIOSLog` is `true` and not do any logging here if that is the case. Otherwise we will duplicate some log lines.

**Comment by jlipps** [comment]
@imurchie what do you think about these changes

File: lib/webdriveragent.js [hunk 2]
```diff
@@ -175,16 +176,20 @@ class WebDriverAgent {
...
-    });
+    if (!this.iosLogAlreadyShown) {
+      logs.on('output', (stdout, stderr) => {
+        let out = stdout || stderr;
+        // make sure we are not reading logs from before this test run
+        if (!loggingStarted && !shouldStartLogging(out)) {
+          return;
+        }
+        if (isPertinentLogLine(out)) {
+          for (let line of out.split("\n").filter(Boolean)) {
```
Comment (imurchie) [inline lib/webdriveragent.js]: Does `.filter(Boolean)` remove everything that is falsy?

Comment (jlipps) [inline lib/webdriveragent.js]: indeed! it's a nice concise way of doing that

Comment (imurchie) [inline lib/webdriveragent.js]: Nice!

**Comment by imurchie** [comment]
Looks good to me!

**Comment by jlipps** [comment]
ok, will wait for CI then merge

**Pull request merged**